### PR TITLE
Add new facet to SFO finder

### PR DESCRIPTION
**Pull request by GDSNewt**
Adds an additional facet to the new to SFO Finder specialist finder. As per documentation found [here](https://docs.publishing.service.gov.uk/repos/specialist-publisher/creating-editing-and-removing-specialist-document-types-and-finders.html#editing-a-specialist-document-type)

[First PR](https://github.com/alphagov/specialist-publisher/pull/2835)
[Trello](https://trello.com/c/E7KMeGZo/3034-specialist-finder-sfo-cases)
[Zendesk](https://govuk.zendesk.com/agent/tickets/5958029)

### CODE REVIEW

File: app/models/sfo_case.rb [hunk 1]
```diff
@@ -1,8 +1,10 @@
 class SfoCase < Document
   validates :sfo_case_state, presence: true
+  validates :sfo_case_opened_date, allow_blank: true, date: true
```
Comment (lauraghiorghisor-tw) [inline app/models/sfo_case.rb]: Can you please check with the users if they want to have this as a required field or not? It was not clear to me that it's optional from the Zendesk ticket.

File: lib/documents/schemas/sfo_cases.json [hunk 2]
```diff
@@ -34,6 +34,14 @@
           "value": "closed"
         }
       ]
+    },
+    {
+      "key": "sfo_case_opened_date",
```
Comment (lauraghiorghisor-tw) [inline lib/documents/schemas/sfo_cases.json:39]: Similarly, include these checks in the question to the user, around the 
- display metadata
- filterable
- preposition (this was missing from the schema, not sure what it defaults to if you don't include it)

**Review by lauraghiorghisor-tw — CHANGES_REQUESTED** [review]
LGTM 

Few bits to check with the user.

File: app/models/sfo_case.rb (hunk 1 above)
Comment (GDSNewt) [reply app/models/sfo_case.rb]: Have checked and they want it to be mandatory, have amended the validation

File: lib/documents/schemas/sfo_cases.json (hunk 2 above)
Comment (GDSNewt) [reply lib/documents/schemas/sfo_cases.json:39]: Have checked with the user on the filterable and metadata issue. The preposition defaults to 'Related to'. In this case "Related to after(date)" and "Related to before (date)". 

I will change it to make it clearer it relates to cases opened

File: lib/documents/schemas/sfo_cases.json [hunk 3]
```diff
@@ -34,6 +34,15 @@
...
+      "name": "Opened date",
+      "short_name": "Opened date",
+      "type": "date",
+      "preposition": "opened",
+      "display_as_result_metadata": true,
```
Comment (lauraghiorghisor-tw) [inline lib/documents/schemas/sfo_cases.json]: Reading the user's comment, I think this should be false?

**Review by lauraghiorghisor-tw — APPROVED** [review]

**Pull request merged**